MixinStickerBlockEntity fails on server environments

### Mod Name

Create

### This issue occurs when only Valkyrien Skies, addons, and the mod I have specified are installed and no other mods

- [ ] I have tested this issue and it occurs with only Valkyrien Skies, addons, and the mod I have specified

### Minecraft Version

1.20.1

### Mod Loader

Fabric

### Issue description

My modpack works locally, but when I start a server, I get the exception below.

Specifically it seems that the `playSound` method cannot be found.:
```
Caused by: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: @Shadow method playSound in fabric-vs_addition-common.mixins.json:create.MixinStickerBlockEntity from mod vs_addition was not located in the target class com.simibubi.create.content.contraptions.chassis.StickerBlockEntity. Using refmap fabric-vs_addition-common-refmap.json
```
 
The `@Shadow` targets this method here: https://github.com/Fabricators-of-Create/Create/blob/mc1.20.1/fabric/dev/src/main/java/com/simibubi/create/content/contraptions/chassis/StickerBlockEntity.java#L96-L99 - note that this is annotated `@Environment(EnvType.CLIENT)`

I'm by no means a modding expert, but from the documentation of `@Environment`, it looks like they actually remove these members in a server environment, so it makes sense that it only works in the client. Maybe there is some way to conditionally `@Shadow` a method based on the `@Environment`?

### Issue reproduction

Install VS additions and Create and launch the server.

### Logs

```
[10:25:26] [main/ERROR]: Failed to start the minecraft server
java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'create' at 'com.simibubi.create.Create'!
        at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:403) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:401) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.16.5.jar:?]
        at net.minecraft.server.Main.main(Main.java:112) ~[server-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) ~[fabric-server-mc.1.20.1-loader.0.16.5-launcher.1.0.1.jar:1.0.1]
        Suppressed: java.lang.NoClassDefFoundError: Could not initialize class com.simibubi.create.AllBlockEntityTypes
                at com.simibubi.create.content.kinetics.simpleRelays.AbstractSimpleShaftBlock.getBlockEntityType(AbstractSimpleShaftBlock.java:53) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
                at com.rabbitminers.extendedgears.registry.ExtendedCogwheelsBlocks.addToBlockEntityType(ExtendedCogwheelsBlocks.java:98) ~[extendedgears-2.1.1-1.20.1-0.5.1.f-fabric.jar:?]
                at com.tterrag.registrate.AbstractRegistrate$Registration.lambda$register$0(AbstractRegistrate.java:124) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
                at com.tterrag.registrate.AbstractRegistrate$Registration.register(AbstractRegistrate.java:124) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.AbstractRegistrate.onRegister(AbstractRegistrate.java:217) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.AbstractRegistrate.lambda$register$1(AbstractRegistrate.java:189) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.fabric.RegistryUtil.forAllRegistries(RegistryUtil.java:12) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.AbstractRegistrate.register(AbstractRegistrate.java:188) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.rabbitminers.extendedgears.fabric.ExtendedCogwheelsImpl.onInitialize(ExtendedCogwheelsImpl.java:16) ~[extendedgears-2.1.1-1.20.1-0.5.1.f-fabric.jar:?]
                at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.16.5.jar:?]
                at net.minecraft.server.Main.main(Main.java:112) ~[server-intermediary.jar:?]
                at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) ~[fabric-server-mc.1.20.1-loader.0.16.5-launcher.1.0.1.jar:1.0.1]
        Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of com.simibubi.create.content.contraptions.chassis.StickerBlockEntity failed [in thread "main"]
                at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
                at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
                ... 8 more
        Suppressed: java.lang.NoClassDefFoundError: Could not initialize class com.simibubi.create.AllBlockEntityTypes
                at com.simibubi.create.content.trains.track.TrackBlock.getBlockEntityType(TrackBlock.java:539) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
                at com.railwayteam.railways.registry.CRTrackMaterials.addToBlockEntityType(CRTrackMaterials.java:266) ~[Steam_Rails-1.6.4+fabric-mc1.20.1.jar:?]
                at com.tterrag.registrate.AbstractRegistrate$Registration.lambda$register$0(AbstractRegistrate.java:124) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
                at com.tterrag.registrate.AbstractRegistrate$Registration.register(AbstractRegistrate.java:124) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.AbstractRegistrate.onRegister(AbstractRegistrate.java:217) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.AbstractRegistrate.lambda$register$1(AbstractRegistrate.java:189) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.fabric.RegistryUtil.forAllRegistries(RegistryUtil.java:12) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.tterrag.registrate.AbstractRegistrate.register(AbstractRegistrate.java:188) ~[registrate-fabric-1.3.62-MC1.20.1-ee112783731be492.jar:?]
                at com.railwayteam.railways.fabric.RailwaysImpl.finalizeRegistrate(RailwaysImpl.java:44) ~[Steam_Rails-1.6.4+fabric-mc1.20.1.jar:?]
                at com.railwayteam.railways.Railways.finalizeRegistrate(Railways.java) ~[Steam_Rails-1.6.4+fabric-mc1.20.1.jar:?]
                at com.railwayteam.railways.Railways.init(Railways.java:81) ~[Steam_Rails-1.6.4+fabric-mc1.20.1.jar:?]
                at com.railwayteam.railways.fabric.RailwaysImpl.onInitialize(RailwaysImpl.java:25) ~[Steam_Rails-1.6.4+fabric-mc1.20.1.jar:?]
                at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.game.minecraft.Hooks.startServer(Hooks.java:63) ~[fabric-loader-0.16.5.jar:?]
                at net.minecraft.server.Main.main(Main.java:112) ~[server-intermediary.jar:?]
                at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.16.5.jar:?]
                at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) ~[fabric-server-mc.1.20.1-loader.0.16.5-launcher.1.0.1.jar:1.0.1]
        Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of com.simibubi.create.content.contraptions.chassis.StickerBlockEntity failed [in thread "main"]
                at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
                at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
                ... 8 more
Caused by: java.lang.BootstrapMethodError: java.lang.RuntimeException: Mixin transformation of com.simibubi.create.content.contraptions.chassis.StickerBlockEntity failed
        at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
        ... 7 more
Caused by: java.lang.RuntimeException: Mixin transformation of com.simibubi.create.content.contraptions.chassis.StickerBlockEntity failed
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.5.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
        at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
        ... 7 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
        at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.5.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
        at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
        ... 7 more
Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [fabric-vs_addition-common.mixins.json:create.MixinStickerBlockEntity from mod vs_addition] from phase [DEFAULT] in config [fabric-vs_addition-common.mixins.json] FAILED during APPLY
        at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:638) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:589) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:379) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.5.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
        at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
        ... 7 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: @Shadow method playSound in fabric-vs_addition-common.mixins.json:create.MixinStickerBlockEntity from mod vs_addition was not located in the target class com.simibubi.create.content.contraptions.chassis.StickerBlockEntity. Using refmap fabric-vs_addition-common-refmap.json
        at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachSpecialMethod(MixinPreProcessorStandard.java:436) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachShadowMethod(MixinPreProcessorStandard.java:412) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachMethods(MixinPreProcessorStandard.java:340) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attach(MixinPreProcessorStandard.java:299) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.createContextFor(MixinPreProcessorStandard.java:277) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinInfo.createContextFor(MixinInfo.java:1292) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:203) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:437) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:418) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.15.3+mixin.0.8.7.jar:0.15.3+mixin.0.8.7]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.16.5.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119) ~[fabric-loader-0.16.5.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[?:?]
        at com.simibubi.create.AllBlockEntityTypes.<clinit>(AllBlockEntityTypes.java:523) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at com.simibubi.create.Create.onInitialize(Create.java:110) ~[create-fabric-0.5.1-f-build.1417+mc1.20.1.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:399) ~[fabric-loader-0.16.5.jar:?]
        ... 7 more
```

--- a/common/src/main/java/io/github/xiewuzhiying/vs_addition/mixin/create/MixinStickerBlockEntity.java
+++ b/common/src/main/java/io/github/xiewuzhiying/vs_addition/mixin/create/MixinStickerBlockEntity.java
@@ -6,6 +6,8 @@ import com.simibubi.create.content.contraptions.glue.SuperGlueItem;
 import com.simibubi.create.foundation.blockEntity.SmartBlockEntity;
 import io.github.xiewuzhiying.vs_addition.VSAdditionConfig;
 import io.github.xiewuzhiying.vs_addition.util.TransformUtilsKt;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
 import net.minecraft.core.BlockPos;
 import net.minecraft.server.level.ServerLevel;
 import net.minecraft.world.level.ChunkPos;
@@ -152,15 +154,13 @@ public abstract class MixinStickerBlockEntity extends SmartBlockEntity {
             }
             if(level.isClientSide && ships.size() > 0) {
                 SuperGlueItem.spawnParticles(level, worldPosition, getBlockState().getValue(StickerBlock.FACING), true);
-                playSound(true);
+                playClientSound(true);
             }
             this.needUpdate = false;
         }
         if (this.needUpdate) {
             clearConstraint();
-            if(level.isClientSide) {
-                playSound(false);
-            }
+            playClientSound(false);
             this.needUpdate = false;
         }
     }
@@ -190,7 +190,17 @@ public abstract class MixinStickerBlockEntity extends SmartBlockEntity {
         return state.isAir() || state.getFluidState() != Fluids.EMPTY.defaultFluidState();
     }
 
+    /**
+     * Safe wrapper for 'playSound', which is removed at runtime on fabric servers.
+     */
+    public void playClientSound(boolean attach) {
+        if(level.isClientSide) {
+            playSound(attach);
+        }
+    }
+
+    @Environment(EnvType.CLIENT)
     @Shadow(remap = false)
-    public abstract void playSound(boolean attach);
+    protected abstract void playSound(boolean attach);
 }
 

